%
% FILE NAME:    masterScriptZebrafishGEM.m
%
%
% PURPOSE: This script is for reconstruction of the Zebrafish-GEM, by using
%          the Human-GEM as template and taking in account specie-specific
%          pathways/reactions.
%
%          Annotation files in tsv format are also generated by
%          integrating human and zebrafish-specific annotation information.
%


%% Add path
addpath(genpath('../../Zebrafish-GEM/'));


%% Prepare Zebrafish ortholog pairs and species-specific network

% get ortholog pairs from human to zebrafish
zebrafishOrthologPairs = extractAllianceGenomeOrthologs('human2ZebrafishOrthologs.tsv');

% load species-specific rxns and mets
rxnsToAdd = importTsvFile('zebrafishSpecificRxns.tsv');
metsToAdd = importTsvFile('zebrafishSpecificMets.tsv');


%% Generate Zebrafish-GEM
[zebrafishGEM, speciesSpecNetwork, gapfillNetwork]=updateAnimalGEM(...
    zebrafishOrthologPairs,rxnsToAdd,metsToAdd,'Zebrafish-GEM',true);


%% Update annotations
[rxnAssoc, metAssoc] = updateAnimalAnnotations(zebrafishGEM,rxnsToAdd,metsToAdd);


%% Save annotation into tsv files, and the model into mat, yml, and xml

% sanity check
if isequal(rxnAssoc.rxns, zebrafishGEM.rxns) && isequal(metAssoc.mets, zebrafishGEM.mets)
	fprintf('sanity check passed.\n');
    exportTsvFile(rxnAssoc,'../model/reactions.tsv');
    exportTsvFile(metAssoc,'../model/metabolites.tsv');
end

zebrafishGEM.geneShortNames = zebrafishGEM.genes;
save('../model/Zebrafish-GEM.mat', 'zebrafishGEM');
exportYaml(zebrafishGEM, '../model/Zebrafish-GEM.yml');
zebrafishGEM = annotateGEM(zebrafishGEM,'../model',{'rxn','met'});  % add annotation data to structure
zebrafishGEM.id = regexprep(zebrafishGEM.id,'-','');  % remove dash from model ID since it causes problems with SBML I/O
exportModel(zebrafishGEM, '../model/Zebrafish-GEM.xml');

